openmdao/utils/om.py

```diff
@@ -236,6 +239,75 @@ def _viewconns(prob):
     return _viewconns
 
 
+def _meta_model_parser(parser):
+    """
+    Set up the openmdao subparser for the 'openmdao meta_model' command.
+
+    Parameters
+    ----------
+    parser : argparse subparser
+        The parser we're adding options to.
+    """
+    parser.add_argument('file', nargs=1, help='Python file containing the model.')
+    parser.add_argument('-m', '--pathname', action='store', dest='pathname',
+                        help='pathname of the metamodel component.')
+
+
+def _meta_model_cmd(options):
+    """
+    Return the post_setup hook function for 'openmdao meta_model'.
+
+    Parameters
+    ----------
+    options : argparse Namespace
+        Command line options.
+
+    Returns
+    -------
+    function
+        The post-setup hook function.
+    """
+    def _view_metamodel(prob):
+        Problem._post_setup_func = None
+
+        mm_types = (MetaModelStructuredComp, MetaModelUnStructuredComp)
+
+        pathname = options.pathname
+
+        if pathname:
+            comp = prob.model._get_subsystem(pathname)
+            if comp and isinstance(comp, mm_types):
+                view_metamodel(comp)
+                return
+        else:
+            comp = None
+
+        metamodels = {mm.pathname: mm for
+                      mm in prob.model.system_iter(include_self=True, typ=mm_types)}
+
+        mm_names = list(metamodels.keys())
+        mm_count = len(mm_names)
+
+        if mm_count == 0:
+            print("No Metamodel components found in model.")
+
+        elif mm_count == 1 and not pathname:
+            comp = metamodels[mm_names[0]]
+            view_metamodel(comp)
+
+        else:
+            try_str = "Try one of the following: {}.".format(mm_names)
+
+            if not pathname:
+                raise AttributeError("Metamodel not specified.\n {}".format(try_str))
+            elif not comp:
+                raise AttributeError("Metamodel '{}' not found.\n {}".format(pathname, try_str))
+            else:
+                raise AttributeError("'{}' is not a Metamodel.\n {}".format(pathname, try_str))
+
```

Comment:
just need the `exit()` call here (like in the other examples, e.g. `_viewconns` above) to prevent the model running afterwards

Reply:
Is there a benefit to that instead of an error raise?

Reply:
I actually think that you want to cleanly tell the user that they provided an invalid input in these cases (with a print) rather than present them with a stack dump that you would get by raising an exception.